builder/builder.go

```diff
@@ -1331,37 +1358,37 @@ func (b *Builder) BuildBundles(template *x509.Certificate, privkey *rsa.PrivateK
 	defer timer.WriteSummary(os.Stdout)
 
 	timer.Start("BUILD BUNDLES")
+	var addLocalConf bool
 	if _, err := os.Stat(b.DNFConf); os.IsNotExist(err) {
-		outfile, err := os.Create(b.DNFConf)
+		if err = ioutil.WriteFile(b.DNFConf, []byte(fmt.Sprintf(dnfConfFormat, b.UpstreamURL)), 0644); err != nil {
+			return errors.Wrapf(err, "Failed to create dnf file: %s", b.DNFConf)
+		}
+		if b.RepoDir != "" {
+			addLocalConf = true
+		}
+	} else if err == nil && b.RepoDir != "" {
+		// check if conf file contains local section
+		raw, err := ioutil.ReadFile(b.DNFConf)
 		if err != nil {
-			helpers.PrintError(err)
-			panic(err)
+			return err
 		}
-		defer func() {
-			_ = outfile.Close()
-		}()
-		if b.RepoDir == "" {
-			cmd := exec.Command("m4", "-D", "UPSTREAM_URL="+b.UpstreamURL, b.DNFTemplate)
-			cmd.Stdout = outfile
-			if err = cmd.Run(); err != nil {
-				helpers.PrintError(err)
-				return err
-			}
-		} else {
-			cmd := exec.Command("m4", "-D", "MIXER_REPO",
```

Comment:
Glad to see this going away. We can remove `m4` from the mixer bundle once this is released.